src/sentry/integrations/vercel/uninstall.py

```diff
@@ -60,35 +93,33 @@ def delete(self, request):
             integration.metadata["webhook_id"] = next_config["webhook_id"]
             integration.metadata["installation_id"] = next_config_id
 
+        integration.save()
+
+        # At this point we can consider if len(orgs) > 1. We have already updated the
+        # integration.metadata, but we may not have deleted the OrganizationIntegration
         try:
             OrganizationIntegration.objects.get(
                 organization_id=configuration["organization_id"], integration_id=integration.id
             ).delete()
+
+            organization = Organization.objects.get(id=configuration["organization_id"])
+            create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=integration.id,
+                event=AuditLogEntryEvent.INTEGRATION_REMOVE,
+                actor_label="Vercel User",
+                data={"provider": integration.provider, "name": integration.name},
+            )
         except OrganizationIntegration.DoesNotExist:
-            logger.error(
-                "vercel.uninstall.missing-org-integration",
+            logger.info(
```

Comment:
@MeredithAnya Between the log message and the `create_audit_entry` call, it feels like we have a lot of duplicated code and it makes this function quite long. Wondering if we can refactor it so we have less repetition and the function is shorter.

Reply:
I tried to organize it into the cases (`len(orgs) == 0`, `len(orgs) == 1`, and `len(orgs) > 1`)  cause that seemed to make sense to me,  but i could see if I could maybe pull out deleting the integration/org integration plus audit log?

Reply:
I think that makes sense